### PR TITLE
RHDEVDOCS 6077 reference for tasks available via cluster resolver

### PR DESCRIPTION
**Pull request by mramendi**
Version(s):

please cp to pipelines-docs-1.15

Issue:

RHDEVDOCS 6077

Link to docs preview:

https://77854--ocpdocs-pr.netlify.app/openshift-pipelines/latest/create/remote-pipelines-tasks-resolvers.html#resolver-cluster-tasks-ref_remote-pipelines-tasks-resolvers

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (savitaashture) [inline modules/op-resolver-cluster-tasks-ref.adoc:173]:
```suggestion
.Changes from the `git-cli` ClusterTask
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (savitaashture) [inline modules/op-resolver-cluster-tasks-ref.adoc:256]:
```suggestion
.Changes from the `git-clone` ClusterTask
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (savitaashture) [inline modules/op-resolver-cluster-tasks-ref.adoc:275]:
```suggestion
.Changes from the `kn-apply` ClusterTask
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (savitaashture) [inline modules/op-resolver-cluster-tasks-ref.adoc:345]:
```suggestion
.Changes from the `maven` ClusterTask
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (savitaashture) [inline modules/op-resolver-cluster-tasks-ref.adoc:393]:
```suggestion
.Changes from the `openshift-client` ClusterTask
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (savitaashture) [inline modules/op-resolver-cluster-tasks-ref.adoc:457]:
```suggestion
.Changes from the `s2i-java` ClusterTask
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (savitaashture) [inline modules/op-resolver-cluster-tasks-ref.adoc:575]:
```suggestion
.Changes from the `skopeo-copy` ClusterTask
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (savitaashture) [inline modules/op-resolver-cluster-tasks-ref.adoc:603]:
```suggestion
.Changes from the `tkn` ClusterTask
```

File: snippets/ref-s2i-task.adoc
Comment (savitaashture) [inline snippets/ref-s2i-task.adoc:61]:
```suggestion
.Changes from the `s2i-{lang}` ClusterTask
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (savitaashture) [inline modules/op-resolver-cluster-tasks-ref.adoc:13]: Can we specify all those custom execution image which we removed here :thinking: 

I mean

You cannot specify a custom execution images (BUILDER_IMAGE, gitInitImage, KN_IMAGE, .......) in any of the tasks available in the {pipelines-shortnames} installation namespace. If you want to use a custom execution image, create a copy of the task and replace the image by editing the copy.

**Review by savitaashture — COMMENTED** [review]
Thank you @mramendi for this huge doc changes

**Review by ppitonak — APPROVED** [review]

**Comment by savitaashture** [comment]
LGTM

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (skrthomas) [inline modules/op-resolver-cluster-tasks-ref.adoc:13]: > If you want to use a custom execution image, create a copy of the task and replace the image by editing the copy.

Is there a Procedure to reference here? Seems like a one-liner that may actually be more complex behind the scenes but also idk anything about this so could just be a misinterpretation.

**Review by skrthomas — COMMENTED** [review]
Looking good! A few comments/questions/observations for you to consider.

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (mramendi) [reply modules/op-resolver-cluster-tasks-ref.adoc:13]: Fair point but out of scope. Created https://issues.redhat.com/browse/RHDEVDOCS-6087

**Review by savitaashture — APPROVED** [review]

File: create/remote-pipelines-tasks-resolvers.adoc
Comment (bergerhoffer) [inline create/remote-pipelines-tasks-resolvers.adoc:47]:
```suggestion
You can use the Git resolver to specify a remote pipeline or task from a Git repostory. The repository must contain a YAML file that defines the pipeline or task. The Git resolver can access a repository either by cloning it anonymously or by using the authenticated SCM API.
```
Just saw this was moved from below, not critical, but I don't think "else" is needed

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:9]: This {pipelines-shortnames} isn't being replaced properly, I'm assuming it's supposed to be singular?

```suggestion
An {pipelines-shortname} installation includes a set of standard tasks that you can use in your pipelines. These tasks are located in the {pipelines-shortname} installation namespace, which is normally the `openshift-pipelines` namespace. You can use the cluster resolver to access the tasks.
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:11]: Might be obvious, but wondering if you should say specify which version

```suggestion
`ClusterTask` functionality is deprecated since {pipelines-shortname} 1.10 and is planned for removal in a future release. If your pipelines use `ClusterTasks`, you can re-create them with the tasks that are available from the {pipelines-shortname} installation namespace by using the cluster resolver. However, certain changes are made in these tasks compared to the existing `ClusterTasks`.
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:22]: Not critical, but just wondering if there's a reason to add the attributes sub if you're not replacing anything here?

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:52]: Maybe consider adjusting the column widths to that the description column gets more room than some of the shorter ones

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:68]: Same here about giving more width to the description column. I won't point out any more tables if you could look through the rest in this file as well.

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:93]: I think you need a comma here. And hyphen

```suggestion
The `git-cli` task runs the `git` command-line utility. You can pass the full Git command or several commands to run using the `GIT_SCRIPT` parameter. If the commands need authentication to a Git repository, for example, in order to complete a push, you must supply the authentication credentials.
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:136]:
```suggestion
|`CRT_FILENAME` |Certificate authority (CA) bundle file name in the `ssl-ca-directory`
```
Also nit, but you could bump a few of these that were hard wrapped to another line back on the same line

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:219]:
```suggestion
|`CRT_FILENAME` |Certificate authority (CA) bundle filename on the `ssl-ca-directory`
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:224]:
```suggestion
|`SUBDIRECTORY` |Relative path in the `output` workspace where the task places the Git repository. |`string` |
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:227]:
```suggestion
|`VERBOSE` |Log the executed commands. |`string` |`false`
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:226]: Can workspace be lower?

```suggestion
|`DELETE_EXISTING` |Delete the contents of the default workspace, if they exist, before running the Git operations. |`string` |`true`
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:341]: These descriptions don't end in periods, but I think most of your other descriptions in this file do. Just double check and try to be consistent if possible.

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:388]:
```suggestion
|`MAVEN_MIRROR_URL` |The Maven repository mirror URL |`string` |
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:389]: Per ISG you don't have to use "on which" - you can end in a preposition to avoid an awkward construction

```suggestion
|`SUBDIRECTORY` |The subdirectory within the `source` workspace that the task runs the Maven build on. |`string` |`.`
```

File: snippets/ref-s2i-task.adoc
Comment (bergerhoffer) [inline snippets/ref-s2i-task.adoc:68]: Just wondering if it would be worthwhile to undefine :lang: at the end of this snippet, like we do with the FeatureName attribute in the technology-preview snippet. Maybe it doesn't make sense if you need it for other things, but undefining it is one way to catch errors easier instead of it accidentally bleeding through elsewhere

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:474]:
```suggestion
The `s2i-go` task builds the source code using the S2I Golang builder image, which is available from the {OCP} registry as `image-registry.openshift-image-registry.svc:5000/openshift/golang`.
```

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:500]: Maybe make the comma into a semi-colon? (Other places too if you do)

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (bergerhoffer) [inline modules/op-resolver-cluster-tasks-ref.adoc:579]: Should be hyphenated. A few other places in the PR if you can check as well

```suggestion
Skopeo is a command-line tool for working with remote container image registries, which does not require a daemon or other infrastructure to load and run the images. The `skopeo copy` command copies an image from one remote registry to another, for example, from an internal registry to a production registry. Skopeo supports authorization on image registries using credentials that you provide.
```

**Review by bergerhoffer — COMMENTED** [review]
Nice job - just a handful of nits if you wouldn't mind addressing before merging. Most critical is the attribute not replacing properly.

File: modules/op-resolver-cluster-tasks-ref.adoc
Comment (mramendi) [reply modules/op-resolver-cluster-tasks-ref.adoc:22]: There are replacements in other snippets so I use the same header for uniformity

File: snippets/ref-s2i-task.adoc
Comment (bergerhoffer) [inline snippets/ref-s2i-task.adoc:70]: The syntax to undefine is:

```suggestion
:!lang:
```

But it doesn't seem to be hurting anything right now. In the interest of time, I'm going to merge this PR as is, if you can please update this in a future PR

Comment (mramendi) [reply snippets/ref-s2i-task.adoc:70]: thanks!

**Pull request merged**
